### PR TITLE
add micropython RP2040 i2c driver

### PR DESCRIPTION
**Pull request by oclyke**
**what it do**
adds an i2c driver that is compatible with the RP2040 port of micropython

**depended on by**
[add support for micropython on RP2040 boards](https://github.com/sparkfun/Qwiic_Py/pull/14)

**status**
✅ i2c driver seems to work

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
**__init__.py**
- See comment regarding this file in the qwiic_py pull request. 
- Basically consolidate everything into this file (only one file) and use exception handling to determine what drivers are available for platform evaluation

Use a pattern like this at the start of the file
_drivers=[]
try:
     from .linux_i2c import LinuxI2C
except e:
    pass
else:
    _drivers.append(LinuxI2C)

**MicroPython**
Does this driver need to be specific to the raspberry pi - can't it be generic micropython?

**Naming**
Instead of the rp2040 name, use "pico". This is "platform" name of the RPi microcontrollers - the rp2040 is just initial microcontroller version

**Comment by sfe-SparkFro** [comment]
Updated to be a more generic MicroPython platform driver, not specific to RP2040.

Also updated platform driver imports like suggested to handle cases when other platform drivers are not included.

Will be reviewing this in more detail before merging.

File: qwiic_i2c/__init__.py [hunk 1]
```diff
@@ -63,11 +63,24 @@
...
+class_names = ["LinuxI2C", "CircuitPythonI2C", "MicroPythonI2C"]
+
+# List of platform drivers found on this system
+_drivers = []
+
+# Loop through all supported platform drivers, see if they're included, and
+# append them to the driver list if so
+for i in range(len(sub_modules)):
+	try:
+		full_name = "qwiic_i2c." + sub_modules[i]
```
Comment (gigapod) [inline qwiic_i2c/__init__.py]: You could skip the creation of the full_name variable and just put the RHS of this statement in the call to import.

File: qwiic_i2c/__init__.py [hunk 2]
```diff
@@ -63,11 +63,24 @@
...
-from .i2c_driver 	import I2CDriver
-from .linux_i2c 	import LinuxI2C
-from .circuitpy_i2c import CircuitPythonI2C
+from .i2c_driver import I2CDriver
 
-_drivers = [LinuxI2C, CircuitPythonI2C]
+# All supported platform module and class names
+sub_modules = ["linux_i2c", "circuitpy_i2c", "micropython_i2c"]
+class_names = ["LinuxI2C", "CircuitPythonI2C", "MicroPythonI2C"]
+
```
Comment (gigapod) [inline qwiic_i2c/__init__.py:74]: One thing that you could do here - use a dict - where you have the sub-module the key and the contents be the class name .Something like:
```pythong
canidate_drivers = { "linux_i2c" : "LinuxI2C", 
                                   "circuitpy_i2c" : "CircuitPythonI2C",
                                  "micropython_i2c" : "MicroPythonI2C"}
  for mod, obj  in canidate_drivers.items():
      try:
          sub_module = __import__("qwiic_i2c."+mod)
          _drivers.append(getattr(sub_module, obj)
     except:
           pass

```
Or something like that -

**Pull request merged**